File: src/lib/scala/Slate.Core/src/main/scala/slate/core/tasks/TaskState.scala
/**
<slate_header>
  url: www.slatekit.com
  git: www.github.com/code-helix/slatekit
  org: www.codehelix.co
  author: Kishore Reddy
  copyright: 2016 CodeHelix Solutions Inc.
  license: refer to website and/or github
  about: A Scala utility library, tool-kit and server backend.
  mantra: Simplicity above all else
</slate_header>
  */

package slate.core.tasks

import slate.common.DateTime

/**
 * Keeps track of the state of the running task.
 * @param name        : Name of the task
 * @param lastRunTime : The last time it ran
 * @param hasRun      : Whether this ran at least 1 time
 * @param runCount    : The number of times the task ran
 * @param errorCount  : The total number of errors
 * @param lastResult  : The last task result
 */
case class TaskState(
                        name          : String   = ""             ,
                        lastRunTime   : DateTime = DateTime.now() ,
                        hasRun        : Boolean  = false          ,
                        runCount      : Int      = 0              ,
                        errorCount    : Int      = 0              ,
                        lastResult    : String   = null
                    )
{
}

File: src/lib/kotlin/slatekit-core/src/main/kotlin/slatekit/core/tasks/TaskState.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.core.tasks

import slatekit.common.DateTime

/**
 * Keeps track of the state of the running task.
 * @param name        : Name of the task
 * @param lastRunTime : The last time it ran
 * @param hasRun      : Whether this ran at least 1 time
 * @param runCount    : The number of times the task ran
 * @param errorCount  : The total number of errors
 * @param lastResult  : The last task result
 */
data class TaskState(
        val name: String,
        val lastRunTime: DateTime,
        val hasRun: Boolean,
        val runCount: Int,
        val errorCount: Int,
        val lastResult: String
) {
}
